R/internal.R

```diff
@@ -1308,3 +1308,41 @@ customcolumnnames <- function(modelnames, custom.columns, custom.col.pos,
 print.texregTable <- function(x, ...) {
   cat(x, ...)
 }
+
+# extract coefficients using the broom package
+broom_coefficients <- function(x) {
+  out <- broom::tidy(x)
+  out <- out[, c('term', 'estimate', 'std.error', 'p.value')]
+  return(out)
+}
+
+# extract gof using the broom package
+broom_gof <- function(x) {
+  # extract
+  out <- broom::glance(x)[1, ]
+  out <- data.frame('gof.names' = colnames(out),
+                    'gof' = as.numeric(out),
+                    'gof.decimal' = TRUE,
+                    stringsAsFactors = FALSE)
+  # rename
+  gof_dict <- c(
+                'adj.r.squared' = 'Adj.\ R$^2$',
+                'deviance' = 'Deviance',
+                'df' = 'DF',
+                'df.residual' = 'DF Resid.',
+                'finTol' = 'Tolerance',
+                'isConv' = 'Convergence',
+                'logLik' = 'Log Likelihood',
+                'null.deviance' = 'Deviance (Null)',
+                'p.value' = 'P Value',
+                'r.squared' = 'R$^2$',
+                'sigma' = 'Sigma',
+                'statistic' = 'Statistic'
+                )
+  gof_dict <- gof_dict[names(gof_dict) %in% out$gof.names]
+  idx <- match(names(gof_dict), out$gof.names)
+  out$gof.names[idx] <- gof_dict
+  # output
+  out <- stats::na.omit(out)
+  return(out)
+}
```

Comment:
What is the advantage of having this `broom`-specific code in `internal.R` rather than in the `extract` method?